### DISCUSSION TITLE
Resize image to fit page

### DISCUSSION THREAD

**Discussion by Unmoon** [py-pdf/pypdf#2457] [Q&A]
I have PDF generated by Confluence that has images that are cut off on the right side. The actual image data is there, but it goes outside page box. I'd like to resize this image to fit inside the page so it's fully visible.

I've tried [this answer](https://stackoverflow.com/a/74999124), but that simply resizes the embedded image, not the displayed size (so even if resizing it to 10x10px, it will still be cut off, just showing a few pixels).

**Comment by pubpub-zz** [comment]
this is not the most trivial operation.

Image ares internally resized to a 0,0-1,1 image space coordinates. 
in order to resize image you have 2 options : replace the image with a new rebuilt one where the original image will be rescaled/moved in order to add transparent areas around. This will work only with shrinked image.
2nd option: 
* get the contentstream of the page, 
* parse it in order to find the "Do" operator corresponding to the image/XObject 
* add before this operator a "q" operator to save the context, add a "cm" operator with a b c d e f parameters with the scale to be applied and shifting
* do not forget to add a "Q" operator after the Do in order to remove the applied scaling.
* replace the original content with the modified one with replace_contents

remember that the clipping may need to be modified too
